lib/checkcondition.h

```diff
@@ -118,6 +118,9 @@ class CPPCHECKLIB CheckCondition : public Check {
     void checkPointerAdditionResultNotNull();
 
 private:
+    // The conditions that have been diagnosed
+    std::set<const Token*> mCondDiags;
+    bool diag(const Token* tok, bool insert=true);
```

Comment:
I would like to have some longer and more descriptive name. I have no suggestion right now.. I will think..